packages/combobox/src/Combobox/Combobox.tsx

```diff
@@ -0,0 +1,740 @@
+}: TriggerValueProps<T>) => {
+  const context = React.useContext(ComboboxContext);
+
+  const height = size === "1" ? "6" : size === "2" ? "8" : "10";
+
+  if (context.value && isMultiSelect(context.value)) {
+    const contextValue = context.value as Array<Option>;
+    const layout = context.layout || "truncate";
+    const truncatedLength = contextValue.length - 2;
+    const truncatedLengthStringArray = truncatedLength.toString().split("");
+
+    if (contextValue.length === 0) {
+      return (
+        <Stack h={height} align="center">
+          <TelegraphButton.Text color="gray">
+            {context.placeholder}
+          </TelegraphButton.Text>
+        </Stack>
+      );
+    }
+
+    return (
+      <Stack
+        gap="1"
+        w="full"
+        my="1"
+        wrap={layout === "wrap" ? "wrap" : "nowrap"}
+        align="center"
+        style={{
+          position: "relative",
+          flexGrow: 1,
+        }}
+      >
+        <AnimatePresence initial={false} mode="popLayout">
+          {contextValue.map((v, i) => {
+            if ((layout === "truncate" && i <= 1) || layout === "wrap") {
+              return (
+                <RefToTgphRef key={v.value}>
+                  <TriggerTag {...v} />
+                </RefToTgphRef>
+              );
+            }
+          })}
+        </AnimatePresence>
+        <AnimatePresence>
+          {layout === "truncate" && contextValue.length > 2 && (
+            <Stack
+              as={motion.div}
+              initial={{ opacity: 0 }}
+              animate={{ opacity: 1 }}
+              exit={{ opacity: 0 }}
+              transition={{ duration: 0.2, type: "spring", bounce: 0 }}
+              h="full"
+              pr="1"
+              pl="8"
+              align="center"
+              aria-label={`${truncatedLength} more selected`}
+              style={{
+                position: "absolute",
+                right: 0,
+                backgroundImage:
+                  "linear-gradient(to left, var(--tgph-surface-1) 0 60%, transparent 90% 100%)",
+              }}
+              key="truncated text"
+            >
+              <Text as="span" size="1" style={{ whiteSpace: "nowrap" }}>
+                +
+                <AnimatePresence mode="wait" initial={false}>
+                  {truncatedLengthStringArray.map((n) => (
+                    <Box
+                      as={motion.span}
+                      w="2"
+                      display="inline-block"
+                      initial={{
+                        opacity: 0.5,
+                      }}
+                      animate={{
+                        opacity: 1,
+                      }}
+                      exit={{
+                        opacity: 0.5,
+                      }}
+                      transition={{ duration: 0.1, type: "spring", bounce: 0 }}
+                      key={n}
+                    >
+                      {n}
+                    </Box>
+                  ))}
```

Comment:
Split the characters of the numbers of "+3 more" to smoothly animate them in and out.